composer/utils/dist.py

```diff
@@ -627,6 +629,76 @@ def get_sampler(
     )
 
 
+def get_node_signal_file_name(rng: Optional[random.Random] = None) -> str:
+    """Returns a file name to use for a file based wait within a node.
+
+    The file name will contain a randomly generated string to avoid conflicts.
+    Note: This file name will be the same on each node, so that it can be used for a file based wait.
+
+    Returns:
+        str: The name of the file that will be created to signal the end of a node's training.
+    """
+    if rng is None:
+        rng = random.Random()
+
+    random_string = ''.join(rng.choices(string.ascii_letters + string.digits, k=6))
+    file_name_list = [f'._signal_file_{random_string}']
+    dist.broadcast_object_list(file_name_list, src=0)
+    return file_name_list[0]
```

Comment:
this returns the global rank 0? is that intnded

Reply:
iirc we want different names per node to avoid shared file system cases

Reply:
ah thank you, i waffled on this but couldn't remember if there was a reason why they were supposed to be node specific. will adjust.

Reply:
fixed